flaky `test_local_directory`

```
____________________________ test_local_directory _____________________________

s = <Scheduler 'tcp://127.0.0.1:55342', workers: 0, cores: 0, tasks: 0>

    @gen_cluster(nthreads=[])
    async def test_local_directory(s):
        with tmpfile() as fn:
            with dask.config.set(temporary_directory=fn):
                w = await Worker(s.address)
>               assert w.local_directory.startswith(fn)
E               AssertionError: assert False
E                +  where False = <built-in method startswith of str object at 0x000001C1E395E670>('C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym.')
E                +    where <built-in method startswith of str object at 0x000001C1E395E670> = 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j'.startswith
E                +      where 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j' = <Worker 'tcp://127.0.0.1:55343', status: running, stored: 0, running: 0/2, ready: 0, comm: 0, waiting: 0>.local_directory

distributed\tests\test_worker.py:1373: AssertionError
---------------------------- Captured stdout call -----------------------------
Dumped cluster state to test_cluster_dump\test_local_directory.yaml
---------------------------- Captured stderr call -----------------------------
2022-08-23 11:15:32,173 - distributed.diskutils - ERROR - Failed to remove 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j' (failed in <built-in function scandir>): [WinError 3] The system cannot find the path specified: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j'
2022-08-23 11:15:32,173 - distributed.diskutils - ERROR - Failed to remove 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j' (failed in <built-in function rmdir>): [WinError 2] The system cannot find the file specified: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpyw3554ym\\dask-worker-space\\worker-vv9ala7j'
```

https://github.com/dask/distributed/runs/7967444014?check_suite_focus=true#step:11:1267

See also: #6935

**Notes:**
* So far, this has only appeared on `windows-latest, 3.10`.

--- a/distributed/tests/test_worker.py
+++ b/distributed/tests/test_worker.py
@@ -1366,21 +1366,19 @@ async def test_register_worker_callbacks_err(c, s, a, b):
 
 
 @gen_cluster(nthreads=[])
-async def test_local_directory(s):
-    with tmpfile() as fn:
-        with dask.config.set(temporary_directory=fn):
-            w = await Worker(s.address)
-            assert w.local_directory.startswith(fn)
-            assert "dask-worker-space" in w.local_directory
+async def test_local_directory(s, tmp_path):
+    with dask.config.set(temporary_directory=str(tmp_path)):
+        w = await Worker(s.address)
+        assert w.local_directory.startswith(str(tmp_path))
+        assert "dask-worker-space" in w.local_directory
 
 
 @gen_cluster(nthreads=[])
-async def test_local_directory_make_new_directory(s):
-    with tmpfile() as fn:
-        w = await Worker(s.address, local_directory=os.path.join(fn, "foo", "bar"))
-        assert w.local_directory.startswith(fn)
-        assert "foo" in w.local_directory
-        assert "dask-worker-space" in w.local_directory
+async def test_local_directory_make_new_directory(s, tmp_path):
+    w = await Worker(s.address, local_directory=str(tmp_path / "foo" / "bar"))
+    assert w.local_directory.startswith(str(tmp_path))
+    assert "foo" in w.local_directory
+    assert "dask-worker-space" in w.local_directory
 
 
 @pytest.mark.skipif(not LINUX, reason="Need 127.0.0.2 to mean localhost")

